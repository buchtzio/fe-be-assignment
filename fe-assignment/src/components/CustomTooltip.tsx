import { Tooltip, TooltipProps } from '@mui/material';
import { styled } from '@mui/system';

interface CustomTooltipProps extends TooltipProps {
  className?: string;
}

const CustomTooltip = styled(({ className, ...props }: CustomTooltipProps) => (
  <Tooltip {...props} classes={{ popper: className }} />
))({
  [`& .MuiTooltip-tooltip`]: {
    backgroundColor: 'white',
    color: 'black',
    boxShadow: '0px 0px 10px rgba(0,0,0,0.1)',
    fontSize: '14px',
  },
});

export default CustomTooltip;
