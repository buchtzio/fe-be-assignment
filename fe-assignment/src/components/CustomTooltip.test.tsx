import React from 'react';
import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import CustomTooltip from './CustomTooltip';
import { TooltipProps } from '@mui/material';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import fs from 'fs';
import path from 'path';

const CustomTooltipTestComponent: React.FC<TooltipProps> = (props) => (
  <ThemeProvider theme={createTheme()}>
    <div>
      <CustomTooltip {...props}>
        <span>Hover me</span>
      </CustomTooltip>
    </div>
  </ThemeProvider>
);

describe('CustomTooltip', () => {
  afterEach(() => {
    // Cleanup after each test
    cleanup();

    // Path to the snapshots folder
    const snapshotsFolder = path.join(__dirname, '__snapshots__');
    // Check if the folder exists and delete it recursively
    if (fs.existsSync(snapshotsFolder)) {
      fs.rmSync(snapshotsFolder, { recursive: true, force: true });
    }
  });

  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(
      <CustomTooltipTestComponent title="Test Tooltip">
        <span>Hover me</span>
      </CustomTooltipTestComponent>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the tooltip text on hover', async () => {
    render(
      <CustomTooltipTestComponent title="Test Tooltip">
        <span>Hover me</span>
      </CustomTooltipTestComponent>,
    );

    // Check that the tooltip text is not visible initially
    expect(screen.queryByText('Test Tooltip')).not.toBeInTheDocument();

    // Simulate hover
    const triggerElement = screen.getByText('Hover me');
    fireEvent.mouseOver(triggerElement);

    // Check that the tooltip text is visible
    const tooltip = await screen.findByRole('tooltip');
    expect(tooltip).toBeInTheDocument();
  });
});
