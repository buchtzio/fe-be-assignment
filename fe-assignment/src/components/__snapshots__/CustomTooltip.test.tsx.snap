// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTooltip renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <span
      aria-label="Test Tooltip"
      class=""
      data-mui-internal-clone-element="true"
    >
      Hover me
    </span>
  </div>
</DocumentFragment>
`;
